In [10]:
import pandas as pd


In [11]:
housing = pd.read_csv("~/Downloads/housing.txt", sep='\s+')

In [12]:
housing.head()

Value  Rooms  %Aged  PT-Ratio  %Lower
0  24000    6.6     65      15.3     5.0
1  21600    6.4     79      17.8     9.1
2  34700    7.2     61      17.8     4.0
3  33400    7.0     46      18.7     2.9
4  36200    7.1     54      18.7     5.3

In [13]:
housing.describe()

Value       Rooms       %Aged    PT-Ratio      %Lower
count    506.000000  506.000000  506.000000  506.000000  504.000000
mean   22532.806324    6.238142   68.616601   18.455534   12.672817
std     9197.104087    1.042763   28.138400    2.164946    7.148724
min     5000.000000   -6.200000    3.000000   12.600000    1.700000
25%    17025.000000    5.900000   45.000000   17.400000    6.975000
50%    21200.000000    6.200000   77.500000   19.050000   11.350000
75%    25000.000000    6.600000   94.000000   20.200000   17.025000
max    50000.000000    8.800000  100.000000   22.000000   38.000000

In [14]:
housing.isna().sum()

Value       0
Rooms       0
%Aged       0
PT-Ratio    0
%Lower      2
dtype: int64

We can verify that there are 2 missing values in the "%Lower" column. 

In [15]:
housing[housing['%Lower'].isnull()]

Value  Rooms  %Aged  PT-Ratio  %Lower
36   20000    5.8     61      19.2     NaN
250  24400    6.5     13      19.1     NaN

For row 36, we may also replace the "NaN" with its mean, because other variables of the same row are close to their means. It's harder to assign a replacement for row 250, as the "%Aged" variable is far away from its mean, while other variables are closer to theirs. However, this way of thinking assumes linear relationships amongst the variables, which may or may not be the case. Given the small number of missing values (2 out of 506), it may be best to simply drop them, since their impact on the dataset is more likely to be minimal.

In [16]:
housing.dropna().describe()

Value       Rooms       %Aged    PT-Ratio      %Lower
count    504.000000  504.000000  504.000000  504.000000  504.000000
mean   22534.126984    6.238492   68.742063   18.452778   12.672817
std     9214.302244    1.044586   28.082684    2.168799    7.148724
min     5000.000000   -6.200000    3.000000   12.600000    1.700000
25%    16950.000000    5.900000   45.000000   17.375000    6.975000
50%    21200.000000    6.200000   78.000000   19.000000   11.350000
75%    25000.000000    6.600000   94.000000   20.200000   17.025000
max    50000.000000    8.800000  100.000000   22.000000   38.000000

We can verify dropping the two rows with missing values does not change much of the summary statistics.

In [17]:
housing_no_missing = housing.dropna()
housing[housing['Rooms'] < 0]





Value  Rooms  %Aged  PT-Ratio  %Lower
15  19900   -5.8     57      21.0     8.5
83  22900   -6.2     47      19.0     7.5